papr/papr-trigger.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/python2
+
+'''
+    This script is run by Jenkins papr-trigger-* jobs to convert GitHub events
+    into a new PAPR pod. PAPR will then create a new collection of jobs for
+    each testsuite to be tested.
+'''
+
+# Note we use Python 2 here because we run directly on the master to reduce
+# overhead. TBD whether we want to add a new bc for creating custom base
+# Jenkins S2I images with additional packages.
+
+from __future__ import print_function
+
+import os
+import sys
+import json
+import uuid
+import argparse
+import tempfile
+import subprocess
+
+
+def main():
+
+    args = parse_args()
+    job = generate_papr_job(args)
+    create_papr_job(job)
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--repo', metavar='OWNER/REPO',
+                         help="GitHub repo to test", required=True)
+    parser.add_argument('--expected-sha1', metavar='SHA1',
+                         help="Expected SHA1 of commit to test")
+    parser.add_argument('--suites', metavar='CONTEXT1|CONTEXT2|...',
+                         help="Pipe-separated list of testsuites to run")
+    branch_or_pull = parser.add_mutually_exclusive_group(required=True)
+    branch_or_pull.add_argument('--branch', help="GitHub branch to test",
+                                default="")
+    branch_or_pull.add_argument('--pull', metavar='ID', default="",
+                                help="GitHub pull request ID to test")
+    return parser.parse_args()
+
+
+def generate_papr_job(args):
+    repo_name = args.repo[args.repo.index('/')+1:]
+    target_name = args.branch if args.branch else args.pull
+    uuid_name = uuid.uuid4().hex[:6] # XXX: actually check for collision
```

Comment:
There has to be some out of the box solution for this somewhere.  Not a blocker now though.